test/gateway/Messages.t.sol

```diff
@@ -61,46 +61,52 @@ contract MessagesTest is Test {
         assertEq(decodedCurrency, uint256(currency));
     }
 
-    function testAddTranche() public {
+    function testAddTrancheMoin() public {
         uint64 poolId = 1;
         bytes16 trancheId = bytes16(hex"811acd5b3f17c06841c7e41e9e04cb1b");
         string memory name = "Some Name";
         string memory symbol = "SYMBOL";
         uint8 decimals = 15;
+        uint8 restrictionSet = 2;
         bytes memory expectedHex =
-            hex"040000000000000001811acd5b3f17c06841c7e41e9e04cb1b536f6d65204e616d65000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000053594d424f4c00000000000000000000000000000000000000000000000000000f";
+            hex"040000000000000001811acd5b3f17c06841c7e41e9e04cb1b536f6d65204e616d65000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000053594d424f4c00000000000000000000000000000000000000000000000000000f02";
```

Comment:
We should change the message decoding to support also messages where the restriction set is not included, so the old and new version are supported, for backwards compatibility (otherwise we depend on centrifuge chain upgrading before release).

I think this can be quite easy by on this line: https://github.com/centrifuge/liquidity-pools/blob/0f77a1a008e082494b131e470bce022fcc86e0c8/src/gateway/Messages.sol#L187 Making it something like
```solidity
if (_msg.length > 186) restrictionSet = BytesLib.toUint8(_msg, 186);
```